[29기 이주미] - Westagram_login_main 구현

dlwnal98 opened the pull request:
-인풋에 입력글자 따라서 백그라운드 컬러 변경 가능

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- login 페이지 레이아웃
- login 버튼 활성화
- main 페이지 레이아웃

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
- ID PW 둘 다 한 글자 이상 입력시 LOGIN버튼 활성화



<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 레이아웃
- js 활성화

<br />

## :: 기타 질문 및 특이 사항
- 아직 디팩토링도 하지 못해서 되게 지저분해 보일 수 있지만
감안해서 속성을 어떻게 쓰는 게 더 효율적이고 좋은 코드인지 알고 싶습니다.
아직 제가 헷갈렸던 개념을 다 정리를 못해서 질문들에 대한 답변이 당장은 간결하지만
수정해서 다시 올릴 예정입니다!

dipsiiiiiiiiii commented:
질문 주신 부분은 코드 리뷰 하면서 같이 답변 드리겠습니다!

dipsiiiiiiiiii commented on students/16th/johnnytae/js/login.js:
```diff
@@ -0,0 +1,17 @@
+const userPw = document.getElementsByClassName('userPw')[0];
+
+function activateLogin() {
+  const filled = userId.value && userPw.value;
```
이런 boolean 값을 가진 변수는 주로 is로 시작합니다!
예를 들어 ```isFilled``` 이렇게 해주세요 :)

dipsiiiiiiiiii commented on students/16th/johnnytae/js/login.js:
```diff
@@ -0,0 +1,17 @@
+
+function activateLogin() {
+  const filled = userId.value && userPw.value;
+  if (!filled) {
+    loginButton.disabled = true;
+    loginButton.style.backgroundColor = '#b5dcfc';
+  } else {
+    loginButton.disabled = false;
+    loginButton.style.backgroundColor = '#49aafa';
+  }
```
굿! 이렇게 하실 수도 있고 다음과 같이 하실 수도 있습니다.
```suggestion
    loginButton.disabled = filled ? false: true;
    loginButton.style.backgroundColor = filled ? '#49aafa' : '#b5dcfc';
```

dipsiiiiiiiiii commented on students/16th/johnnytae/js/main.js:
```diff
@@ -0,0 +1,34 @@
+const userComment = document.getElementsByClassName('user_comment')[0];
+
+const addComment = () => {
+  const add = document.createElement('li');
+  const addId = document.createElement('span');
+  const addCmt = document.createElement('span');
+  addId.className = 'user_id';
+  addCmt.className = 'user_comment';
+  addId.textContent = userId.textContent;
+  addCmt.textContent = ` ${commentBody.value}`;
+
+  add.appendChild(addId);
+  add.appendChild(addCmt);
+  commentList.appendChild(add);
```
이렇게 하나씩 createElement 하실 수도 있지만
innerHTML을 활용하면 좀 더 편하게 사용하실 수 있습니다!

dipsiiiiiiiiii commented on students/16th/johnnytae/js/main.js:
```diff
@@ -0,0 +1,34 @@
+};
+
+function activatePost() {
+  if (!commentBody.value) {
+    postButton.disabled = true;
+    postButton.style.color = 'rgb(155, 184, 226)';
+  } else {
+    postButton.disabled = false;
+    postButton.style.color = 'rgb(8, 98, 224)';
+    postButton.style.cursor = 'pointer';
+  }
```
이 부분도 Login.js 파일 리뷰 참고해서 한 번 리팩토링 해보세요!

dipsiiiiiiiiii commented on students/16th/johnnytae/login.html:
```diff
@@ -0,0 +1,30 @@
+</head>
+<body>
+  <div class="wrapper">
+    <div>
```
여기 div는 필요 없을 것 같네요!

dipsiiiiiiiiii commented on students/16th/johnnytae/main.html:
```diff
@@ -0,0 +1,102 @@
+</head>
+<body>
+  <nav>
+    <div class="navWrap">
```
여기 div도 필요 없을 것 같습니다!

dipsiiiiiiiiii commented on students/16th/johnnytae/main.html:
```diff
@@ -0,0 +1,102 @@
+
+  <main class="body-wrap">
+    <div class="feed-wrap">
+      <div class="story-wrap">
+        stories
+        <ul class="stories">
```
ul 안에 stories 넣으시면 div 없어도 될 것 같네요!

dipsiiiiiiiiii commented on students/16th/johnnytae/style/common.css:
```diff
@@ -0,0 +1,3 @@
+* {
+  box-sizing: border-box;
+}
```
이런 초기화 속성은 reset.css에 넣어 주세요!

dipsiiiiiiiiii commented on students/16th/johnnytae/style/main.css:
```diff
@@ -0,0 +1,240 @@
+body {
+  padding: 0;
+  margin: 0;
```
이런 초기화 속성은 reset.css에 넣어 주세요!

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다!
div를 아주 조금 불필요하게 사용하고 계시지만 그래도 태그 잘 활용하고 계시고 있는 것 같습니다 :)
리뷰 반영해주세요!

Commit:
-버튼 백그라운드 컬러 변경 기능 추가
-인풋에 입력글자 따라서 백그라운드 컬러 변경 가능

Commit:
login 페이지 레이아웃 + 로그인 버튼 활성화 구현
main 페이지 레이아웃

Commit:
리팩토링
1. semantic tag로 구성 : 로고 부분을 h1태그로 변경.
2. 클래스 이름 변경 : 추상적으로 짓거나 단순히 번호, 순서를 매겨서 이름을 지었었는데 좀 더 의미있게, 잘 캐치할 수 있게 이름을 변경함.
3. top-up 방식으로 변경 : 아직 코딩이 서툴러서 큰 틀을 잡기위해 부모 요소에 크기를 줬었는데 자식 요소의 크기를 조절해서 top-up 방식으로 변경함.
